lib/matplotlib/backends/backend_pgf.py

```diff
@@ -990,4 +1009,217 @@ def _cleanup_all():
+    ...     fig = plt.figure()
+    ...     pdf.savefig(fig)
+    ...     # When no figure is specified the current figure is saved
+    ...     pdf.savefig()
+    """
+    __slots__ = (
+        '_outputfile',
+        'keep_empty',
+        '_tmpdir',
+        '_basename',
+        '_fname_tex',
+        '_fname_pdf',
+        '_n_figures',
+        '_file',
+        'metadata',
+    )
+
+    def __init__(self, filename, *, keep_empty=True, metadata=None):
+        """
+        Create a new PdfPages object.
+
+        Parameters
+        ----------
+
+        filename : str
+            Plots using :meth:`PdfPages.savefig` will be written to a file at
+            this location. Any older file with the same name is overwritten.
+        keep_empty : bool, optional
+            If set to False, then empty pdf files will be deleted automatically
+            when closed.
+        metadata : dictionary, optional
+            Information dictionary object (see PDF reference section 10.2.1
+            'Document Information Dictionary'), e.g.:
+            `{'Creator': 'My software', 'Author': 'Me',
+            'Title': 'Awesome fig'}`
+
+            The standard keys are `'Title'`, `'Author'`, `'Subject'`,
+            `'Keywords'`, `'Producer'`, `'Creator'` and `'Trapped'`.
+            Values have been predefined for `'Creator'` and `'Producer'`.
+            They can be removed by setting them to the empty string.
+        """
+        self._outputfile = filename
+        self._n_figures = 0
+        self.keep_empty = keep_empty
+        self.metadata = metadata or {}
+
+        # create temporary directory for compiling the figure
+        self._tmpdir = tempfile.mkdtemp(prefix="mpl_pgf_pdfpages_")
+        self._basename = 'pdf_pages'
+        self._fname_tex = os.path.join(self._tmpdir, self._basename + ".tex")
+        self._fname_pdf = os.path.join(self._tmpdir, self._basename + ".pdf")
+        self._file = open(self._fname_tex, 'wb')
+
+    def _write_header(self, width_inches, height_inches):
+        supported_keys = {
+            'title', 'author', 'subject', 'keywords', 'creator',
+            'producer', 'trapped'
+        }
+        infoDict = {
+            'creator': 'matplotlib %s, https://matplotlib.org' % __version__,
+            'producer': 'matplotlib pgf backend %s' % __version__,
+        }
+        metadata = {k.lower(): v for k, v in self.metadata.items()}
+        infoDict.update(metadata)
+        hyperref_options = ''
+        for k, v in infoDict.items():
+            if k not in supported_keys:
+                raise ValueError(
+                    'Not a supported pdf metadata field: "{}"'.format(k)
+                )
+            hyperref_options += 'pdf' + k + '={' + str(v) + '},'
+
+        latex_preamble = get_preamble()
+        latex_fontspec = get_fontspec()
+        latex_header = r"""\PassOptionsToPackage{{
+  {metadata}
+}}{{hyperref}}
+\RequirePackage{{hyperref}}
```

Comment:
any reason why it's not a standard "usepackage[...]{hyperref}"? (perhaps, I dunno)

Reply:
Because People might use hyperref in their rc params latex header and like this it avoids the "hyperref loaded with incompatible options" error

Reply:
so that's equivalent to "if hyperref is already loaded by the preamble, then merge these options with those passed in the preamble"?  (again, just asking)

Reply:
Yes, and the user specified options take precedence.